### DISCUSSION TITLE
theorem-title not in its own HTML paragraph

### DISCUSSION THREAD

**Discussion by zeileis** [quarto-dev/quarto-cli#1925] [Q&A] [answered]
When using one of the [theorem divs](https://quarto.org/docs/authoring/cross-references.html#theorems-and-proofs), then in the HTML output the corresponding "theorem-title" is sometimes in its own paragraph and sometimes not. Specifically, if the div just consists of a single paragraph, then the title is in a separate paragraph. Otherwise it is merged into the first paragraph.

As a concrete example consider:

```
::: {#exr-1}
Compute: 25 + 17.
:::

::: {#exr-2}
Let a = 25 and b = 17.

Compute: a + b.
:::
```

I have simply inserted the snippet above into the `intro.qmd` from the `mybook` demo (`quarto create-project mybook --type book`). This yields the following HTML output using quarto 1.0.38 with pandoc 2.17.1.1 on Debian GNU/Linux.

```
<div id="exr-1" class="theorem exercise">
<p><span class="theorem-title"><strong>Exercise 1.1 </strong></span></p>
<p>Compute: 25 + 17.</p>
</div>
<div id="exr-2" class="theorem exercise">
<p><span class="theorem-title"><strong>Exercise 1.2 </strong></span>Let a = 25 and b = 17.</p>
<p>Compute: a + b.</p>
</div>
```

Rendered in a browser it looks like this:

![browser screenshot](https://i.imgur.com/GJWj8k8.png)

Note that the text of the first exercise starts in a new paragraph while the second exercise starts in the same paragraph as the title.

Is this difference by design? Can I somehow switch between these layouts? I would like a consistent layout, preferably with the theorem-title in its own paragraph.

**Comment by mcanouil** [comment]
Somehow, the content of the div is not taken as a whole because of the breakline, e.g., the following  examples work properly (to note, with or without level 2 heading which is the part in parentheses next to the label produce the same behaviour):

- Using `</br>` HTML tag. (one or two)
  ```
  ::: {#exr-2}
  
  Let a = 25 and b = 17.
  </br></br>
  Compute: a + b.
  
  :::
  ```

- With two trailing spaces
  ```
  ::: {#exr-2}
  
  Let a = 25 and b = 17.  
  Compute: a + b.
  
  :::
  ```

**Reply by zeileis** [reply]
Thanks for the suggestion. It's a workaround but not ideal because with the two trailing spaces I just get a line break (`<br>` in HTML and `\\` in LaTeX) but not a new paragraph which I would really need here for the intended layout.

**Comment by cscheid** [comment]
I believe this to be by design. Theorems are often titled in math papers and those titles tend to appear inline next to the theorem name.

I agree that it would be useful to have an option of an "unnamed" theorem, but I'm not sure that it exists right now.

**Reply by mcanouil** [reply]
What's the difference with the level 2 header that goes into parentheses in bold and what you are calling name that goes in normal text next to the bold part?

**Reply by zeileis** [reply]
Thanks for the feedback, very much appreciated!

However, I thought that theorem name or label should be given with `##` in an extra line according to the [documentation](https://quarto.org/docs/authoring/cross-references.html#theorems-and-proofs):

```
::: {#exr-1}
## 42a
Compute: 25 + 17.
:::

::: {#exr-2}
## 42b
Let a = 25 and b = 17.

Compute: a + b.
:::
```

This includes the `##` title in brackets in the "theorem-title" span:

```
<div id="exr-1" class="theorem exercise">
<p><span class="theorem-title"><strong>Exercise 1.1 (42a) </strong></span></p>
<p>Compute: 25 + 17.</p>
</div>
<div id="exr-2" class="theorem exercise">
<p><span class="theorem-title"><strong>Exercise 1.2 (42b) </strong></span>Let a = 25 and b = 17.</p>
<p>Compute: a + b.</p>
</div>
```
Rendered in the browser:

![browser screenshot](https://i.imgur.com/pSm0mOY.png)

Therefore, the theorem name/label seems to be orthogonal to the paragraph handling.

**Reply by mcanouil** [reply]
@zeileis This was exactly my understanding.
(#1926)

**Reply by zeileis** [reply]
Yes, thanks! I hadn't seen your follow-up before completing mine. I didn't delete it because I thought that it shows the issue even more verbosely.

**Comment by zeileis** [comment, marked as answer]
Just for the record, this has been fixed by: https://github.com/quarto-dev/quarto-cli/issues/1926

Thanks everyone for your help!
